pkg/controller/plan/tls_utils.go

```diff
@@ -50,23 +51,29 @@ func ComposeTLSSecret(namespace, clusterName, componentName string) (*v1.Secret,
 		SetStringData(map[string]string{}).
 		GetObject()
 
-	const tpl = `{{- $cert := genCA "KubeBlocks" 36500 }}
-{{ $cert.Cert }}
-{{ $cert.Key }}
-`
-	out, err := buildFromTemplate(tpl, nil)
+	// use ca gen cert
+	// IP: 127.0.0.1
+	// DNS: localhost and *.<clusterName>-<componentName>-headless.<namespace>.svc.cluster.local
+	SignedCertTpl := fmt.Sprintf(`
+	{{- $ca := genCA "KubeBlocks" 36500 -}}
+	{{- $cert := genSignedCert "%s peer" (list "127.0.0.1") (list "localhost" "*.%s-%s-headless.%s.svc.cluster.local") 36500 $ca -}}
+	{{- $ca.Cert -}}
+	{{- print "___spliter___" -}}
+	{{- $cert.Cert -}}
+	{{- print "___spliter___" -}}
+	{{- $cert.Key -}}
+`, componentName, clusterName, componentName, namespace)
+	out, err := buildFromTemplate(SignedCertTpl, nil)
 	if err != nil {
 		return nil, err
 	}
-	index := strings.Index(out, "-----BEGIN RSA PRIVATE KEY-----")
-	if index < 0 {
+	parts := strings.Split(out, "___spliter___")
+	if len(parts) != 3 {
 		return nil, errors.Errorf("wrong cert format: %s", out)
```

Comment:
error msg should be more user friendly. something like this: "generate TLS certificates failed with cluster name xxx, component name xxx".
and don't dump the whole cert as the private key is rather sensitive.

Reply:
sure, but maybe dump the whole cert not a bad thing for debug, cause this reconcile will fail and cert will become useless?